keyboards/aki27/cocot46plus/cocot46plus.c

```diff
@@ -0,0 +1,286 @@
+#endif
+
+
+cocot_config_t cocot_config;
+uint16_t cpi_array[] = COCOT_CPI_OPTIONS;
+uint16_t scrl_div_array[] = COCOT_SCROLL_DIVIDERS;
+uint16_t angle_array[] = COCOT_ROTATION_ANGLE;
+#define CPI_OPTION_SIZE (sizeof(cpi_array) / sizeof(uint16_t))
+#define SCRL_DIV_SIZE (sizeof(scrl_div_array) / sizeof(uint16_t))
+#define ANGLE_SIZE (sizeof(angle_array) / sizeof(uint16_t))
+
+
+// Trackball State
+bool     BurstState        = false;  // init burst state for Trackball module
+uint16_t MotionStart       = 0;      // Timer for accel, 0 is resting state
+
+// Scroll Accumulation
+static int16_t h_acm       = 0;
+static int16_t v_acm       = 0;
+
+
+void pointing_device_init_kb(void) {
+    // set the CPI.
+    pointing_device_set_cpi(cpi_array[cocot_config.cpi_idx]);
+    adns5050_write_reg(0x22, 0b10000 | 0x80);
+}
+
+
+report_mouse_t pointing_device_task_kb(report_mouse_t mouse_report) {
+
+    double rad = angle_array[cocot_config.rotation_angle] * (M_PI / 180) * -1;
+    int8_t x_rev =  + mouse_report.x * cos(rad) - mouse_report.y * sin(rad);
+    int8_t y_rev =  + mouse_report.x * sin(rad) + mouse_report.y * cos(rad);
+
+    if (cocot_get_scroll_mode()) {
+        // rock scroll direction
+        if (abs(x_rev) > abs(y_rev)) {
+            y_rev = 0;
+        } else {
+            x_rev = 0;
+        }
+
+        // accumulate scroll
+        h_acm += x_rev * cocot_config.scrl_inv;
+        v_acm += y_rev * cocot_config.scrl_inv * -1;
+
+        int8_t h_rev = h_acm >> scrl_div_array[cocot_config.scrl_div];
+        int8_t v_rev = v_acm >> scrl_div_array[cocot_config.scrl_div];
+
+        // clear accumulated scroll on assignment
+
+        if (h_rev != 0) {
+            if (mouse_report.h + h_rev > 127) {
+                h_rev = 127 - mouse_report.h;
+            } else if (mouse_report.h + h_rev < -127) {
+                h_rev = -127 - mouse_report.h;
+            }
+            mouse_report.h += h_rev;
+            h_acm -= h_rev << scrl_div_array[cocot_config.scrl_div];
+        }
+        if (v_rev != 0) {
+            if (mouse_report.v + v_rev > 127) {
+                v_rev = 127 - mouse_report.v;
+            } else if (mouse_report.v + v_rev < -127) {
+                v_rev = -127 - mouse_report.v;
+            }
+            mouse_report.v += v_rev;
+            v_acm -= v_rev << scrl_div_array[cocot_config.scrl_div];
+        }
+
+        mouse_report.x = 0;
+        mouse_report.y = 0;
+    } else {
+        mouse_report.x = x_rev;
+        mouse_report.y = y_rev;
+    }
+
+    return pointing_device_task_user(mouse_report);
+}
+
+
+
+bool process_record_kb(uint16_t keycode, keyrecord_t* record) {
+    // xprintf("KL: kc: %u, col: %u, row: %u, pressed: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed);
+    
+    if (!process_record_user(keycode, record)) return false;
+
+    switch (keycode) {
+#ifndef MOUSEKEY_ENABLE
+        // process KC_MS_BTN1~8 by myself
```

Comment:
> process KC_MS_BTN1~8 by myself

Whats the added value here? It seems to just be duplicating core.

Reply:
I'm not sure why that was there. Seems to work OK without it. Removing.